drivers/hyperv/hyperv.go

```diff
@@ -271,39 +275,37 @@ func (d *Driver) Create() error {
 }
 
 func (d *Driver) chooseVirtualSwitch() (string, error) {
-	if d.VSwitch == "" {
-		// Default to the first external switche and in the process avoid DockerNAT
-		stdout, err := cmdOut("[Console]::OutputEncoding = [Text.Encoding]::UTF8; (Hyper-V\\Get-VMSwitch -SwitchType External).Name")
-		if err != nil {
-			return "", err
-		}
-
-		switches := parseLines(stdout)
+	if strings.ToLower(d.VSwitch) == strings.ToLower(dockerNATSwitch) {
+		return "", fmt.Errorf("vswitch %s cannot be used", dockerNATSwitch)
```

Comment:
Can we also tell the user why we cannot use this switch?

Also, instead of checking this Docker switch specifically, let's make sure that we allow only `External` switches and not the `Internal` & `Private` switch. That way, the check will be more generalized.

Reply:
I just changed the logic back to original one which distinguished the cases on whether vSwitch name is specified.

* When the vSwitch name is not specified, the driver uses either external vswitches or the default NAT vSwitch `Default Switch`, and the NAT vSwitch `Default Switch` wins. If the NAT vSwitch doesn't exist (perhaps for those users who is still running before 1709), they could get the behavior as like previous one.
* When the vSwitch name is specified, the driver uses the vSwitches whose names match, and `External` one wins (the descending order makes sure that as `External` switch type 2 > `Internal` 1 > `Private` 0)

From Hyper-V perspective, I don't think here we should limit the switch to only `External` as it could make sense to Docker VM to communicate with the host or other VMs through the `Internal` or `Private` vSwitch, but only when the vSwitch name is specified.